File: binary-search/binary-search.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        hi = len(nums)-1
        lo = 0
        while lo <= hi:
            mid = (hi + lo) // 2
            if nums[mid] == target:
                return mid
            elif nums[mid] < target:
                lo = mid+1
            else:
                hi = mid-1
        return -1

File: binary-search/binary-search.java
class Solution 
{
    public int search(int[] nums, int target) 
    {
        if (nums.length < 1)
            return -1;
        int low = 0, high = nums.length-1;
        
        while (low <= high)
        {
            int mid = (int) ((low + high) / 2);
            if (nums[mid] == target)
                return mid;
            else if (nums[mid] > target)
                high = mid-1;
            else
                low = mid+1;
        }
        
        return -1;
    }
}